instrumentation/src/com/intellij/rt/coverage/instrumentation/LineEnumerator.java

```diff
@@ -189,16 +188,17 @@ public void visitLookupSwitchInsn(Label dflt, int[] keys, Label[] labels) {
   }
 
   public void visitTableSwitchInsn(int min, int max, Label dflt, Label[] labels) {
-    super.visitTableSwitchInsn(min, max, dflt, labels);
-    if (!myHasExecutableLines) return;
-    final LineData lineData = myClassInstrumenter.getLineData(myCurrentLine);
-    if (lineData != null) {
-      rememberSwitchLabels(dflt, labels);
-      SwitchData switchData = lineData.addSwitch(myCurrentSwitch++, min, max);
-      mySwitchLabels.put(dflt, switchData);
+    if (myHasExecutableLines) {
+      final LineData lineData = myClassInstrumenter.getLineData(myCurrentLine);
+      if (lineData != null) {
+        rememberSwitchLabels(dflt, labels);
+        SwitchData switchData = lineData.addSwitch(myCurrentSwitch++, min, max);
+        mySwitchLabels.put(dflt, switchData);
+      }
+      myState = SEEN_NOTHING;
+      myHasInstructions = true;
     }
-    myState = SEEN_NOTHING;
-    myHasInstructions = true;
+    super.visitTableSwitchInsn(min, max, dflt, labels);
```

Comment:
it seems counter-intuitive that we call super at the end here, but not in other methods

Reply:
I fixed filters order, so `super` call position does not depend on filters. Anyway `super` calls in `visitJump` are still unusual as we have to substitute labels. The same reason will lead to `super` call disorder in switch methods after #34 merge